samcli/local/layers/layer_downloader.py

```diff
@@ -36,7 +36,11 @@ def __init__(self, layer_cache, cwd, lambda_client=None):
         """
         self._layer_cache = layer_cache
         self.cwd = cwd
-        self.lambda_client = lambda_client or boto3.client('lambda')
+        self._lambda_client = lambda_client
+
+    @property
+    def lambda_client(self):
+        return self._lambda_client or boto3.client('lambda')
```

Comment:
I'm ok with the change, just curious if it will change our failure mode scenarios anywhere.

Reply:
No. This just means we create the default boto3 client when the property is accessed instead of on creation. This will mean we will be creating a new boto3 client on every access though. Maybe this is a even better way?

```
 self._lambda_client = self._lambda_client or boto3.client('lambda')
 return self._lambda_client
```